[Python] Document `failed_request_status_codes` option better (#11598)
getsentry/sentry-docs: docs/platforms/python/integrations/fastapi/index.mdx

  A `set` of integers that will determine which status codes should be reported to Sentry.

>   The `failed_request_status_codes` option only controls whether [`HTTPException`](https://fastapi.tiangolo.com/reference/exceptions/?h=httpexception) exceptions should be
>   reported to Sentry. Other unhandled exceptions, which do not have a `status_code` attribute, are always reported to Sentry.
> 

Reviewer: ```suggestion
  The `failed_request_status_codes` option determines whether [`HTTPException`](https://fastapi.tiangolo.com/reference/exceptions/?h=httpexception) exceptions should be
  reported to Sentry. Unhandled exceptions that don't have a `status_code` attribute will always be reported to Sentry.

```
